Toplean_Alexandru/week-02/Domain/Domain/CreateEmployeeOp/CreateEmployeeOp.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Domain.Models;
+using Infrastructure.Free;
+using LanguageExt;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using static Domain.Domain.CreateEmployeeOp.CreateEmployeeResult;
+using static Domain.Models.Employee;
+
+namespace Domain.Domain.CreateEmployeeOp
+{
+    public class CreateEmployeeOp : OpInterpreter<CreateEmployeeCmd, ICreateEmployeeResult, Unit>
+    {
+        public override Task<ICreateEmployeeResult> Work(CreateEmployeeCmd Op, Unit state)
+        {
+            if (Exists(Op.Name))
+                return Task.FromResult<ICreateEmployeeResult>(new EmployeeNotCreated(EmployeeErrorCode.NameExists)); // Employee already exists
+            else
+            {
+                if (Op.IsValid().Item1)
```

Comment:
Please use deconstruction instead of accessing elements in tuple via the Item properties: https://docs.microsoft.com/en-us/dotnet/csharp/deconstruct. Code is much more clear when you use deconstruction.